californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -248,7 +235,7 @@ public boolean registerOutboundRequest(final Exchange exchange) {
 			throw new IllegalArgumentException("exchange does not contain a request");
 		} else {
 			int mid = registerWithMessageId(exchange, exchange.getCurrentRequest());
-			if (mid > Message.NONE) {
+			if (Message.NONE != mid) {
```

Comment:
My idea of changing that, was to have a defined value range for `mid` and a defined usage pattern.
There is code, which uses `Message.NONE` just as values (therefore `!=` or `==`) and others as the negative value (therefore `>`). There was even a method (`MessageIdTracker.getNextMessageId()`), which returns `-1`. 
So I replaced the `-1` and go for `!=` or `==`.